core/templates/dev/head/pages/exploration-editor-page/improvements-tab/services/improvement-modal.service.ts

```diff
@@ -48,6 +48,177 @@ angular.module('oppia').factory('ImprovementModalService', [
+               * no further expansion is possible.
+               * @returns {boolean}
+               */
+              $scope.canExpandActions = function() {
+                return indexOfFirstDisplayedAction !== 0;
+              };
+
+              /**
+               * Iterates through the learner actions and expands the set of
+               * actions to be displayed. If all learner actions have been
+               * handled, no more expansion is possible.
+               */
+              let expandActions = function() {
+                let i;
+                let previousStateName = null;
+                if (indexOfFirstDisplayedAction <
+                    playthrough.actions.length - 1) {
+                  let action = playthrough.actions[
+                    indexOfFirstDisplayedAction];
+                  previousStateName =
+                      action.actionCustomizationArgs.state_name.value;
+                }
+                for (i = indexOfFirstDisplayedAction; i >= 0; i--) {
+                  let action = playthrough.actions[i];
+                  let currentStateName =
+                      action.actionCustomizationArgs.state_name.value;
+
+                  if (currentStateName !== previousStateName) {
+                    // When the latest state name and the state name of the
+                    // current learner action differ, they are part of
+                    // different contexts. So, we identify whether the size
+                    // of the block has crossed the threshold number of actions
+                    // per block. If it has, we don't add more learner actions
+                    // to the block.
+                    if (indexOfFirstDisplayedAction - i >= (
+                      MAX_UNRELATED_ACTIONS_PER_BLOCK)) {
+                      indexOfFirstDisplayedAction = i + 1;
+                      break;
+                    }
+                  }
+                  previousStateName = currentStateName;
+                }
+                // This is the case when all learner actions have been iterated
+                // over, and no more expansion is possible.
+                if (i === -1) {
+                  indexOfFirstDisplayedAction = 0;
+                }
+              };
+
+              expandActions();
+
+              // Index to know where to start highlighting actions.
+              let indexOfActionInFirstDisplayedBlock =
+                indexOfFirstDisplayedAction;
+
+              $scope.issueIsMultipleIncorrectSubmissions = false;
+              if (playthrough.issueType === (
+                ISSUE_TYPE_MULTIPLE_INCORRECT_SUBMISSIONS)) {
+                $scope.issueIsMultipleIncorrectSubmissions = true;
+              }
+
+              /**
+               * Renders a table for MultipleIncorrectSubmissions issue for a
+               * more friendly UI.
+               */
+              $scope.renderIssueTable = function() {
```

Comment:
Just to check -- this is shown additional to the regular actions display, not replacing it, right?

Reply:
Nope, it replaces a set of learner actions. The first display block which has multiple answer submit actions are displayed as a screen instead of a set of learner actions.

Reply:
Can you help me understand this a bit more? I see a `ng-if="issueIsMultipleIncorrectSubmissions"` but no corresponding `ng-if="!issueIsMultipleIncorrectSubmissions"` condition for the other part of the display.

Reply:
Discussed this. We don't require the other ng-if because it was handled in getActionsToRender method